mpas_analysis/shared/climatology/climatology.py

```diff
@@ -608,17 +806,23 @@ def ds_to_weights(ds):
             for var in ds.data_vars:
                 weights[var] = ds[var].notnull()
         else:
-            raise TypeError('ds must be an instance of either xarray.Dataset or xarray.DataArray.')
+            raise TypeError('ds must be an instance of either xarray.Dataset '
+                            'or xarray.DataArray.')
 
         return weights
 
-    dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+    if maskVaries:
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
 
-    weights = ds_to_weights(ds)
+        weights = ds_to_weights(ds)
 
-    weightSum = (weights * ds.daysInMonth).sum(dim='Time')
+        weightSum = (weights * ds.daysInMonth).sum(dim='Time')
 
-    timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+        timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+    else:
+        days = ds.daysInMonth.sum(dim='Time')
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+        timeMean = dsWeightedSum / days.where(days > 0.)
```

Comment:
If the mask doesn't vary with time, you definitely still want to take into account the fact that different months have different lengths in computing your climatology.  The else case is the whole point of having the `maskVaries` flag.  You yourself complained that this code had slowed down the computation by a factor of 2.  So I added the ability to *not* bother with the mask in the denominator if it will always either be masked out or not.

There is, in my estimation, no case where `ds.mean()` is appropriate because it always matters that months have different lengths and that we store data averaged over monthly intervals.

I've added more to the docstrings of the non-private functions that have a `maskVaries` flag to try to clarify when this might matter.